fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/BaseKeyExpression.java

```diff
@@ -102,7 +102,8 @@ public boolean isPrefixKey(@Nonnull KeyExpression key) {
 
             if (prefixPart instanceof AtomKeyExpression && keyPart instanceof AtomKeyExpression) {
                 // there should be an atom comparison here
-                if (((AtomKeyExpression) prefixPart).equalsAtomic((AtomKeyExpression) keyPart)) {
+                boolean equals = forSort ? prefixPart.equivalentForSort(keyPart) : ((AtomKeyExpression) prefixPart).equalsAtomic((AtomKeyExpression) keyPart);
```

Comment:
Note that `equalsAtomic()` deliberately compares only the information at a particular node in the `KeyExpression` tree and _doesn't_ look at children. I think that `equivalentForSort()` does, so is this really what we want to use here?

Reply:
Are there `AtomKeyExpression`s that have children?

Reply:
Yes. For example, `NestingKeyExpression`, `SplitKeyExpression`, and `FunctionKeyExpression`.

Reply:
Right. I guess I meant are there other implementations of `equalsAtomic` that obey that rule that they don't look at their children. Those do include them.

Reply:
I reverted the addition of a parameter to `isPrefixKey`. To then make it work with the new planner, I made `FieldKeyExpression.equalsAtomic` be `equivalentForSort`. This makes the general contract a little special, but this is the only use currently.